src/web/tools/GestureHandlerOrchestrator.ts

```diff
@@ -66,7 +66,20 @@ export default class GestureHandlerOrchestrator {
     return hasToWait;
   }
 
+  private shouldBeCancelledByFinishedHandler(handler: GestureHandler): boolean {
+    return this.gestureHandlers.some(
+      (otherHandler) =>
+        this.shouldHandlerWaitForOther(handler, otherHandler) &&
+        otherHandler.getState() === State.END
+    );
+  }
+
   private tryActivate(handler: GestureHandler): void {
+    if (this.shouldBeCancelledByFinishedHandler(handler)) {
```

Comment:
As far as I remember this change was necessary, but I'll check that.

Reply:
Okay, I've looked into it - we have to keep this change. However, we don't have to use pointers cache on android since this change seems to work even without those caches (I know that's `.ts` file, but it is the same change as the one in android).